pkg/admiralctl/migrate/migrate.go

```diff
@@ -0,0 +1,694 @@
+
+	name string
+	gvk  schema.GroupVersionKind
+}
+
+// AddFlags adds flags to the specified FlagSet.
+func (o *CommandMigrateOption) AddFlags(flags *pflag.FlagSet) {
+	flags.BoolVar(&o.AutoCreatePolicy, "auto-create-policy", true,
+		"Automatically create a PropagationPolicy for namespace-scoped resources or "+
+			"create a ClusterPropagationPolicy for cluster-scoped resources.")
+	flags.StringVar(&o.PolicyName, "policy-name", "",
+		"The name of the PropagationPolicy(or ClusterPropagationPolicy) that is automatically created after migration. "+
+			"If not specified, the name will be the resource name with a hash suffix that is generated by resource metadata.")
+	flags.StringVarP(&o.OutputFormat, "output", "o", "", "Output format. One of: json|yaml")
+	flags.StringVar(&o.SchedulingMode, "scheduling-mode", string(fedcorev1a1.SchedulingModeDuplicate),
+		"determines the mode used by the scheduler when scheduling federated objects, One of: Duplicate|Divide, default is Duplicate.")
+	flags.StringVarP(&o.Namespace, "namespace", "n", o.Namespace, "If present, the namespace scope for this CLI request.")
+	flags.StringVarP(&o.Cluster, "cluster", "C", "", "the name of legacy cluster (eg -C=member1)")
+	flags.StringVar(&o.ClusterContext, "cluster-context", "",
+		"Context name of legacy cluster in kubeconfig. Only works when there are multiple contexts in the kubeconfig.")
+	flags.StringVar(&o.ClusterKubeConfig, "cluster-kubeconfig", "",
+		"Path of the legacy cluster's kubeconfig.")
+	flags.BoolVar(&o.DryRun, "dry-run", false, "Run the command in dry-run mode, without making any server requests.")
+}
+
+// Preflight ensures that options are valid and marshals them if necessary
+func (o *CommandMigrateOption) Preflight(f util.Factory, args []string) error {
+	var err error
+
+	if len(args) != 2 {
+		return fmt.Errorf("incorrect command format, please use correct command format")
+	}
+
+	o.name = args[1]
+
+	if o.OutputFormat == "yaml" || o.OutputFormat == "json" {
+		o.Printer = func(mapping *meta.RESTMapping, outputObjects *bool,
+			withNamespace bool, withKind bool,
+		) (printers.ResourcePrinterFunc, error) {
+			printer, err := o.JSONYamlPrintFlags.ToPrinter(o.OutputFormat)
+
+			if genericclioptions.IsNoCompatiblePrinterError(err) {
+				return nil, err
+			}
+
+			printer, err = printers.NewTypeSetter(scheme.Scheme).WrapToPrinter(printer, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			return printer.PrintObj, nil
+		}
+	}
+
+	if o.Namespace == "" {
+		o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+		if err != nil {
+			return fmt.Errorf("failed to get namespace from Factory. error: %w", err)
+		}
+	}
+
+	// If '--cluster-context' not specified, take the cluster name as the context.
+	if len(o.ClusterContext) == 0 {
+		o.ClusterContext = o.Cluster
+	}
+
+	if o.Cluster == "" {
+		return fmt.Errorf("the cluster cannot be empty")
+	}
+
+	if o.ClusterKubeConfig == "" {
```

Comment:
Why member cluster kubeconfig is required? We can obtain the access config from FederatedCluster

Reply:
I haven't tried this successfully before, I'll take another look at it.